framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,298 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TimedSubdomainModifier.h"
+#include "TimedElementSubdomainModifier.h"
+#include "DelimitedFileReaderOfString.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseApp", TimedSubdomainModifier);
+
+InputParameters
+TimedSubdomainModifier::validParams()
+{
+  InputParameters params = TimedElementSubdomainModifier::validParams();
+
+  // parameters for direct input (additionally to 'times')
+  params.addParam<std::vector<std::string>>("blocks_from", "Names or ids of the 'old' block.");
+  params.addParam<std::vector<std::string>>("blocks_to", "Names or ids of the 'new' block.");
+  params.addParamNamesToGroup("times blocks_from blocks_to", "Direct data input.");
+
+  // parameters for file-based data supply
+  params.addParam<FileName>("data_file", "File holding CSV data");
+  params.addParam<bool>("header",
+                        "Indicates whether the file contains a header with the column names");
+  params.addParam<std::string>("delimiter", ",", "Delimiter used to parse the file");
+  params.addParam<std::string>("comment", "#", "Comment character used to parse the file");
+  params.addParam<size_t>(
+      "time_column_index", 0, "Zero-based index of the time column. Default is '0'.");
+  params.addParam<size_t>(
+      "blocks_from_column_index", 1, "Zero-based index of the blocks_from column. Default is '1'.");
+  params.addParam<size_t>(
+      "blocks_to_column_index", 2, "Zero-based index of the blocks_to column. Default is '2'.");
+  params.addParam<std::string>("time_column_text", "Header text of the time column.");
+  params.addParam<std::string>("blocks_from_column_text", "Header text of the blocks_from column.");
+  params.addParam<std::string>("blocks_to_column_text", "Header text of the blocks_to column.");
+  params.addParamNamesToGroup(
+      "data_file header delimiter comment time_column_index blocks_from_column_index "
+      "blocks_to_column_index time_column_text blocks_from_column_text blocks_to_column_text",
+      "Data input from file.");
+
+  params.addClassDescription("Moves all elements of a block to another at the given times.");
+
+  return params;
+}
+
+TimedSubdomainModifier::TimedSubdomainModifier(const InputParameters & parameters)
+  : TimedElementSubdomainModifier(parameters)
+{
+  // determine function arguments
+
+  int bFromData_File =
```

Comment:
```suggestion
 const auto from_data_file =
```
style guide. apply to the others pls